app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/ImageCompressor.kt

```diff
@@ -0,0 +1,113 @@
+package com.simplemobiletools.smsmessenger.helpers
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Matrix
+import android.media.ExifInterface
+import android.net.Uri
+import com.simplemobiletools.commons.extensions.getCompressionFormat
+import com.simplemobiletools.commons.extensions.getMyFileUri
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.smsmessenger.extensions.extension
+import com.simplemobiletools.smsmessenger.extensions.getExtensionFromMimeType
+import com.simplemobiletools.smsmessenger.extensions.isImageMimeType
+import java.io.File
+import java.io.FileOutputStream
+
+/**
+ * Compress image to a given size based on
+ * [Compressor](https://github.com/zetbaitsu/Compressor/)
```

Comment:
this looks like a maintained library, maybe we can just include the whole thing?

Reply:
The latest version of the library uses coroutines. Shall I proceed to integrate it?

Reply:
but we have to use coroutines too?

Reply:
Yes. We have to.  Unless we use an older version

Reply:
ok nevermind then, it isnt worth spending too much time on